### PR TITLE
txn: make prewrite idempotent when falling back to 2PC

### PR DESCRIPTION
**Pull request by youjiali1995**
Signed-off-by: youjiali1995 <zlwgx1023@gmail.com>

### What problem does this PR solve?

Issue Number: close https://github.com/pingcap/tidb/issues/23477

Problem Summary:

`Prewrite` is not idempotent when falling back to 2PC:
1. A async-commit prewrite request succeeded in writing a fallback lock. Its response is `min_commit_ts == 0`.
2. Following duplicated prewrite requests returns `min_commit_ts != 0` which makes clients think it can go on with async-commit protocol.

### What is changed and how it works?

Proposal: [xxx](url) 

What's Changed:

Make prewrite idempotent.

### Related changes
- Need to cherry-pick to the release branch

### Check List 

Tests 

- Unit test
- Integration test

Side effects

### Release note 
* No release note.

### CODE REVIEW

File: src/storage/txn/commands/prewrite.rs [hunk 1]
```diff
@@ -1074,25 +1075,28 @@ mod tests {
...
-            false,
-            Context::default(),
-        );
-
-        let res = prewrite_command(&engine, cm, &mut statistics, cmd).unwrap();
-        assert!(res.min_commit_ts.is_zero());
-        assert!(res.one_pc_commit_ts.is_zero());
-        assert!(!must_locked(&engine, k1, 20).use_async_commit);
-        assert!(!must_locked(&engine, k2, 20).use_async_commit);
+        // Test the idempotency of prewrite when falling back to 2PC.
```
Comment (cfzjywxk) [inline src/storage/txn/commands/prewrite.rs:1081]: How about the case that, for a prewrite request with `k1` and `k2`, the `max_ts` calculated for `k1` is ok, while the `max_ts` of the concurrency manager changed before the process of `k2` and the prewrite of `k2` fallbacked to 2pc.
Then if the prewrite request succeded and the request is retried, it's needed to verify the `min_commit_ts` in the response is zero.

**Comment by youjiali1995** [comment]
@cfzjywxk @MyonKeminta PTAL

**Comment by cfzjywxk** [comment]
LGTM

**Review by MyonKeminta — APPROVED** [review]
LGTM

**Review by cfzjywxk — APPROVED** [review]

**Review by sticnarf — APPROVED** [review]
LGTM

**Pull request merged**